packages/@lwc/integration-karma/test/template/directive-if-elseif-else/index.spec.js

```diff
@@ -148,22 +149,126 @@ describe('lwc:if, lwc:elseif, lwc:else directives', () => {
+                verifyExpectedSlotContent(child, true);
+            });
+        });
+
+        it('should properly rerender content for slots', () => {
+            const element = createElement('x-parent', { is: XparentWithSlot });
+            element.condition = true;
+            document.body.appendChild(element);
+
+            const child = element.shadowRoot.querySelector('x-child-with-slot');
+
+            verifyExpectedSlotContent(child, true);
+
+            element.condition = false;
+            return Promise.resolve()
+                .then(() => {
+                    verifyExpectedSlotContent(child, false);
+
+                    element.condition = true;
+                })
+                .then(() => {
+                    verifyExpectedSlotContent(child, true);
+                });
+        });
+
+        describe('named slots', () => {
+            /**
+             * Utility function to verify that slot content is assigned.
+             *
+             * @param {Element} child Child element to verify.
+             * @param {Boolean} condition Whether slot content is expected or not expected.
+             */
+            function verifyExpectedNamedSlotContent(child, condition) {
                 const assignedNodes = child.shadowRoot.querySelector('slot').assignedNodes();
-                expect(assignedNodes).toHaveSize(3); // VFragment has empty text nodes as delimiters
-                expect(assignedNodes[1].innerHTML).toBe('Named slot content from parent');
+                if (condition) {
+                    expect(assignedNodes.length).toBe(3);
+                    expect(assignedNodes[0].innerHTML).toBe('static slot header');
+                    expect(assignedNodes[1].innerHTML).toBe('conditional slot content');
+                    expect(assignedNodes[2].innerHTML).toBe('static slot footer');
+                } else {
+                    expect(assignedNodes.length).toBe(2);
+                    expect(assignedNodes[0].innerHTML).toBe('static slot header');
+                    expect(assignedNodes[1].innerHTML).toBe('static slot footer');
+                }
+            }
+
+            it('should properly assign content for named slots', () => {
+                const element = createElement('x-parent', { is: XparentWithNamedSlot });
+                document.body.appendChild(element);
+
+                const child = element.shadowRoot.querySelector('x-child-with-named-slot');
+
+                // When if condition is false, no slot content is provided by parent
+                verifyExpectedNamedSlotContent(child, false);
+                element.condition = true;
+                return Promise.resolve().then(() => {
+                    verifyExpectedNamedSlotContent(child, true);
+                });
+            });
+
+            xit('should not override default slot content when no elements are explicitly passed to the default slot', () => {
```

Comment:
Included a test case for the VFragment's text delimiters overriding default slot content to illustrate the behavior.

Reply:
Because with the current behavior, it overrides the default slot with the text nodes.